commit 818e4d60f121e4db8d98d4bcd11cd34644753291
Author: Trent Mick <trentm@gmail.com>
Date:   2019-04-22T15:25:55-07:00 (6 months ago)
    
    OS-7762 vm runtest/runtests do not list new cores if running without a terminal
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/src/vm/runtest b/src/vm/runtest
index 326fc38c..d856f985 100755
--- a/src/vm/runtest
+++ b/src/vm/runtest
@@ -205,11 +205,6 @@ if [[ ${TEST_EXIT_CODE} != 0 ]]; then
     tests="?"
 fi
 
-coredir=$(cd $(dirname $(coreadm $$ | awk '{gsub("%Z", "/", $2); print $2;}')) \
-    && pwd -P)
-cores=$(find "$coredir" -type f -newer /tmp/runtest.start.$$)
-rm /tmp/runtest.start.$$
-
 if [[ -t 0 ]]; then
     # We're on a terminal so output the summary
     echo "#"
@@ -224,15 +219,24 @@ if [[ -t 0 ]]; then
     if [[ ${fail} -gt 0 ]]; then
         echo -e "# \033[31mFAIL: ${fail} / ${tests}\033[39m"
     fi
-    if [[ -n "$cores" ]]; then
+    echo "#"
+fi
+
+coredir=$(cd $(dirname $(coreadm $$ | awk '{gsub("%Z", "/", $2); print $2;}')) \
+    && pwd -P)
+cores=$(find "$coredir" -type f -newer /tmp/runtest.start.$$)
+rm /tmp/runtest.start.$$
+
+if [[ -n "$cores" ]]; then
+    if [[ -t 0 ]]; then
         echo -e "# \033[31mNew core files\033[39m"
-        # Send a list of core files to stdout for testers and `runtests` to see
-        # and another in this run's log.
-        for core in $cores; do
-            echo "# new core: $core"
-        done | tee -a /tmp/test.output.$$
     fi
-    echo "#"
+
+    # Send a list of core files to stdout for testers and `runtests` to see
+    # and another in this run's log.
+    for core in $cores; do
+        echo "# new core: $core"
+    done | tee -a /tmp/test.output.$$
 fi
 
 if [[ ${fail} -gt 0 || -n "$cores" ]]; then
